services/src/main/java/org/keycloak/forms/login/freemarker/FreeMarkerLoginFormsProvider.java

```diff
@@ -217,14 +217,20 @@ public Response createResponse(UserModel.RequiredAction action) {
     protected Response createResponse(LoginFormsPages page) {
 
         Theme theme;
+        Locale locale;
         try {
             theme = getTheme();
         } catch (IOException e) {
             logger.error("Failed to create theme", e);
             return Response.serverError().build();
         }
+        if (user == null && this.authenticationSession != null) {
+            locale = session.getContext().resolveLocale(this.authenticationSession.getAuthenticatedUser());
+        }
+        else{
+            locale = session.getContext().resolveLocale(user);
+        }
```

Comment:
@rishabhsvats There is `setUser` method in this class  to set the user and, checking other handlers, it is called (for example here in [VerifyEmailActionTokenHandler](https://github.com/keycloak/keycloak/blob/26.0.6/services/src/main/java/org/keycloak/authentication/actiontoken/verifyemail/VerifyEmailActionTokenHandler.java#L85). So better if, instead of adding this change, you just call this method in the `ExecuteActionsActionTokenHandler`:

```diff
diff --git a/services/src/main/java/org/keycloak/authentication/actiontoken/execactions/ExecuteActionsActionTokenHandler.java b/services/src/main/java/org/keycloak/authentication/actiontoken/execactions/ExecuteActionsActionTokenHandler.java
index 17bb665acc..d8b4b4bb26 100644
--- a/services/src/main/java/org/keycloak/authentication/actiontoken/execactions/ExecuteActionsActionTokenHandler.java
+++ b/services/src/main/java/org/keycloak/authentication/actiontoken/execactions/ExecuteActionsActionTokenHandler.java
@@ -90,6 +90,7 @@ public class ExecuteActionsActionTokenHandler extends AbstractActionTokenHandler

             return session.getProvider(LoginFormsProvider.class)
                     .setAuthenticationSession(authSession)
+                    .setUser(authSession.getAuthenticatedUser())
                     .setSuccess(Messages.CONFIRM_EXECUTION_OF_ACTIONS)
                     .setAttribute(Constants.TEMPLATE_ATTR_ACTION_URI, confirmUri)
                     .setAttribute(Constants.TEMPLATE_ATTR_REQUIRED_ACTIONS, token.getRequiredActions())
```

I see there are no tests for this, so just adding this line would be OK.